Hand-in | 1. december 2021

thepiotrek opened the pull request:
As requested, just a pull request from developer to master.

bwesth commented on src/components/Navbar.css:
```diff
@@ -0,0 +1,24 @@
+}
+
+a:link {
+  color: #34401A;
+  text-decoration: none;
+  font-size: 24px;
+  padding-left: 36px;
+  padding-right: 36px;
```
Looks like you're using your design system. Great!
If you want to use .css files, I suggest looking into CSS variables.
Will make your life easier, and improve your code quality.

bwesth commented on src/components/Navbar.js:
```diff
@@ -1,15 +1,42 @@
-        </div>
-    )
-}
+class Navbar extends Component {
```
Why class components all the sudden?
I'm curious.

bwesth commented on src/components/Navbar.js:
```diff
@@ -1,15 +1,42 @@
+
+    render() {
 
+        if (this.state.user == "none") {
+            return (<div className="Navbar">
+                <Link to="/">Start</Link>
+            </div>)
```
Another way of doing this is to redirect to a login screen if there is no user. You're on the right path thinking in conditional rendering, but it looking into how other websites are doing it might inspire you. Try to log out of your favourite sites and see how it changes.

== compared non-strictly. Unless you have a good reason for not doing it, use === by default.

Instead of user = "none" you can have user = false or null. Then you can test if (user) instead.

bwesth commented on src/components/Navbar.js:
```diff
@@ -1,15 +1,42 @@
+            </div>)
+        } else if (this.state.user == "participant") {
+            return (
+                <div className="Navbar">
+                    <Link to="/excursions">Excursion</Link>
+                    <Link to="/transport">Transportation</Link>
+                    <Link to="/contact">Contact</Link>
+                    <Link to="/signup">Sign up</Link>
+                    <Link to="/">Start</Link>
+                </div>
+            )
+        } else if (this.state.user == "organizer") {
+            return (
+                <div className="Navbar">
+                    <Link to="/excursions">Excursion</Link>
+                    <Link to="/participantList">Participant List</Link>
+                    <Link to="/dutyList">Duty List</Link>
+                    <Link to="/shoppingList">Shopping List</Link>
+                    <Link to="/">Start</Link>
+                </div>
```
Don'tRepeatYourself :)
Limit your differences between two boolean branches (if-else) to only what actually differs. Here you have the containing "NavBar" div, Excursion, and Start links on both branches.

On a different notion. Do your organizers not need to also sign up for excursions? Currently, if the user is organizer they have no access to the sign up page :)

bwesth commented on src/pages/contact.js:
```diff
@@ -2,10 +2,11 @@ import React from "react"
 const Contact = () => {
     return (
-        <div>
+        <div className="General">
```
CSS classes are per convention kebab-case. This means they should be lower case, and with a dash (-) in between each word.
There are some people who name CSS classes **for components** with upper case first letter (like .App, or in this case, .Contact). But this is not universal.

bwesth commented on src/pages/excursions.js:
```diff
@@ -1,87 +1,84 @@
+	return (
+		<div className='Container'
+			style={{
+				backgroundColor: '#B4BF5E',
+				marginTop: '-21px',
+				marginLeft: '282px',
+				marginRight: '282px',
+				height: '180vh'
```
Don't mix styles. Doing it this way is fine (although I suggest looking into "React styled components"), but don't use both this and seperate CSS files :)

bwesth commented on src/pages/excursions.js:
```diff
@@ -1,87 +1,84 @@
+						fontSize: 28
+					}}
+				>
+					Friday to sunday<strong> June 7-9 2021</strong><br />Location: <strong>Vilshult in Småland, SWEDEN</strong><br />Price: Adult 950 kr.<br />  Child 475 kr.
```
I get this is a work in progress, but I highly recommend putting all your "dummy"/test data in one place. a "testData.js" file for instance, and then import it. Hard-coding it like you do here gives you more work later on.

bwesth commented on src/pages/excursions.js:
```diff
@@ -1,87 +1,84 @@
-};
+					}}
+				>
+					Join this year’s excursion, there is room for 150 people. So sign up before your colleague </p>
```
A good example of what I wrote about hard-coding. You might forget to fix this. I could easily forget, at least.
In "Room for 150 people" 150 might change from excursion to excursion. If you put these variables in data instead of text, you'll remember to fix it later.

bwesth commented on src/pages/participantList.js:
```diff
@@ -0,0 +1,20 @@
+
+    return (
+        <div className="General">
+            <h3>Name: | Duty Prefrences: | Age:</h3> 
+            {participantList}
```
I was taught never to use tables unless I actually _need_ a table. This is one of those :)

bwesth commented on src/pages/shoppingList.js:
```diff
@@ -0,0 +1,15 @@
+import React from "react"
+
+const ShoppingList = () => {
```
This way is fine, but for the final version, do it the same way in all functional components. Consistency 👍

bwesth commented on src/tempData/participants.js:
```diff
@@ -0,0 +1,30 @@
+const tempParData = [
+    {
+        pid: 1,
+        name: "Anne Nielsen",
+        birthDate: 23,
+        dutyPreference: "Kitchen, Cleaning, Cooking"
+    },
```
Great! :D

bwesth reviewed (commented):
Good first push

SaraTolstrupHusum commented on src/index.js:
```diff
@@ -1,17 +1,22 @@
+  
+  
+  
+//  API KEY: "y1pkFZswJRDe8jhue98PGNh4mxxu5geUNCvNKXK1"
+//   JS_KEY: "HrNmgVEuGQ8macNhM9BbXfkORi4wJOr5HFFks8O6"
```
Leaving your keys in comments defeats the purpose of putting them in the .env-files. ;)

thepiotrek replied:
just leftovers from creating the env files on each individual machine ;)

SaraTolstrupHusum commented on src/pages/excursions.js:
```diff
@@ -1,87 +1,65 @@
+	return (
+		<div className='container'>
 
+			<h1 className='about_h1' style={{
```
An `<h1>`-tag named "about_**h1**" might be a bit "No shit, Sherlock". ;)
And why add the className and the do inline styling? Now I am utterly confused. ;p
You use style sheets. So don't do inline styling all of a sudden.

SaraTolstrupHusum commented on src/pages/excursions.js:
```diff
@@ -1,87 +1,65 @@
+			<div className='first-collumn-image'>
+				<img src={Cabin}
+					alt="sweden"
+					style={{
+						justifyContent: "center",
+						position: "absolute",
+						width: "50%",
+						height: "50%",
+						left: "402px",
+						top: "257px",
+						alignItems: 'center'
+					}}
```
Inline styling again!!! Please stop.

SaraTolstrupHusum commented on src/components/Navbar.js:
```diff
@@ -1,15 +1,41 @@
-const Navbar = () => {
+export default function Navbar() {
+
+    const [currentUser] = useState(Parse.User.current().getUsername())
```
I suspect you don't really need the useState here. You can just make a const.
`const currentUser = Parse.User.current().getUsername()`

SaraTolstrupHusum commented on src/pages/Home.js:
```diff
@@ -0,0 +1,48 @@
+    
+
+    return (
+        <div className="General">
```
Hmm... General might not be that descriptive.

SaraTolstrupHusum reviewed (approved):
Your file naming is not consistent. Some of the file names for your pages are with lower case but your imports are upper case. Component should be PascalCase and so should their file names. I could not get your code to compile before I renamed A LOT of paths. This is a problem you **have to** fix. Your code should compile after an `npm install`.
Apart from the little hick-up above. You are working well with separating your components and using descriptive names for your variables. You have some way to go still, but just keep at it. Come ask us about our comments if they don't make sense.

thepiotrek commented on src/App.js:
```diff
@@ -1,34 +1,39 @@
-import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
-import './App.css'
+import { BrowserRouter, Routes, Route } from 'react-router-dom'
+// import './App.css'
```
A question/comment til the file naming:

- the problem is a result of trying to fix the consistency issues, but the case of the letters is not registered by Github. We were fixing that all .js files should start with a capital letter (going from small ones), and it is the case on our machines, but not in Github. 

Do you know a good fix to that?

thepiotrek replied:
Okay. nevermind... 

'git mv' for the win!

thepiotrek commented on src/components/TransportCard.js:
```diff
@@ -0,0 +1,18 @@
+
+export default function TransportCard(props){
+    return (<div className="card">
+            <p className="card--heading"> {props.par.name} </p>
+            <img alt="" className="card--picture" src={props.par.img}/>  
+            <p className="card--text">Car color: <b>{props.par.carColor}</b> </p> 
+            <p className="card--text"> License plate:<b> {props.par.licensePlate} </b></p>
+            <p className="card--text"> Seats:<b> {props.par.seats}</b></p>  
+            <p className="card--text"> Departure:<b> {props.par.depature} </b></p>
+            <p className="card--button"><GreenButton><ButtonText> Reserve seat</ButtonText></GreenButton>
+            <ButtonDisabled><ButtonText> Cancel seat</ButtonText></ButtonDisabled></p> 
+            </div>
```
The double dashes are inspired by the current React tutorial on Scrimba. I quite like it - here it starts with the general styling for 'card' and then the subpart come after '--' 
Do you think it is wrong?

thepiotrek commented:
@bwesth @SaraTolstrupHusum @mircealungu 

We fixed a lot based on your comments.
And added new pages & components.
There is more communication with Back 4 App now.
We are looking forward to your comments, especially as we are moving towards "the end" of the course.
We could definitely use some guidelines on, what we should focus on in the last weeks:

More functionality?
Better styling?
Something else? 

Cheers!
